numpy-stubs/__init__.pyi

```diff
@@ -368,6 +368,58 @@ class ndarray(_ArrayOrScalarCommon, Iterable, Sized, Container):
     def flatten(self, order: str = ...) -> ndarray: ...
     def ravel(self, order: str = ...) -> ndarray: ...
     def squeeze(self, axis: Union[int, Tuple[int, ...]] = ...) -> ndarray: ...
+    # Operations
+    @overload
+    def all(
+        self, axis: None = ..., out: None = ..., keepdims: Optional[bool] = ...
+    ) -> bool_: ...
+    @overload
+    def all(
```

Comment:
Note that specifying an axis doesn't necessarily mean you will get an array out; e.g.

```python
>>> x = np.array([True, True])
>>> x.all(axis=0)
True
```

And the analogue with more axes:

```python
>>> x = np.ones((2, 2), dtype=np.bool_)
>>> x.all(axis=(0, 1))
True
```

If we had `Literal` we could make that guarantee for when `keepdims = True`, but we can't do that yet.